ci/docker/debian-experimental-cpp.dockerfile

```diff
@@ -111,13 +112,13 @@ RUN /arrow/ci/scripts/install_sccache.sh unknown-linux-musl /usr/local/bin
 # Prioritize system packages and local installation.
 ENV ARROW_ACERO=ON \
     ARROW_AZURE=ON \
-    ARROW_BUILD_TESTS=ON \
+    ARROW_BUILD_TESTS=OFF \
     ARROW_DATASET=ON \
     ARROW_DEPENDENCY_SOURCE=SYSTEM \
     ARROW_DATASET=ON \
     ARROW_FLIGHT=ON \
     ARROW_FLIGHT_SQL=ON \
-    ARROW_GANDIVA=ON \
+    ARROW_GANDIVA=OFF \
```

Comment:
Can we use newer LLVM instead?

```diff
diff --git a/dev/tasks/tasks.yml b/dev/tasks/tasks.yml
index 9f04d33f83..fb7d5a4ce5 100644
--- a/dev/tasks/tasks.yml
+++ b/dev/tasks/tasks.yml
@@ -1011,6 +1011,7 @@ tasks:
         ARCH: "amd64"
         DEBIAN: "experimental"
         GCC: "15"
+        LLVM: "19"
       flags: "-e CMAKE_CXX_STANDARD=20"
       image: debian-cpp
 
```

Reply:
I'll try!

Reply:
Sorry... This didn't help...

Reply:
Ah yeah, for posterity / anyone else reading:

```
/usr/include/c++/15/ciso646:46:4: error: #warning "<ciso646> is deprecated in C++17, use <version> to detect implementation-specific macros" [-Werror=cpp]
   46 | #  warning "<ciso646> is deprecated in C++17, use <version> to detect implementation-specific macros"
      |    ^~~~~~~
cc1plus: all warnings being treated as errors
```

I think this is ok for now. I'm preparing the small thrift patch for CRAN. We don't build tests or Gandiva there so we won't get hit with these. We can deal with them (or ideally they are dealt with upstream...) when GCC 15 is actually released